File: Examples/ros_mscl_cpp_example/src/listener.cpp
#include "ros/ros.h"
#include "sensor_msgs/Imu.h"

void imuDataCallback(const sensor_msgs::Imu::ConstPtr& imu)
{
	ROS_INFO("Quaternion Orientation:    [%f, %f, %f, %f]", imu->orientation.x, imu->orientation.y, imu->orientation.z, imu->orientation.w);
	ROS_INFO("Angular Velocity:          [%f, %f, %f]", imu->angular_velocity.x, imu->angular_velocity.y, imu->angular_velocity.z);
	ROS_INFO("Linear Acceleration:       [%f, %f, %f]", imu->linear_acceleration.x, imu->linear_acceleration.y, imu->linear_acceleration.z);
	
	// add code here to handle incoming IMU data
}

int main(int argc, char** argv)
{
	// register listener to ROS master
	ros::init(argc, argv, "listener");

	// get the device name parameter
	std::string deviceName;
	ros::NodeHandle params("~");
	params.param<std::string>("device", deviceName, "gx5");
	ROS_INFO("Got device param: %s", deviceName.c_str());

	// clear param for future use
	params.deleteParam("device");

	// create the listener node object
	ros::NodeHandle n;

	// subscribe to the imu/data topic
	//Parameters:
	//   topic - namespace (defined in launch file) and topic name
	//   queue size - maximum number of messages to buffer
	//   callback - callback function to handle this data
	ros::Subscriber sub = n.subscribe(("/" + deviceName + "/imu/data"), 3, imuDataCallback);

	// start listening for data
	ros::spin();

	return 0;
}

File: Examples/ros_mscl_py_example/src/listener/listener.py
#! /usr/bin/env python
import rospy
from sensor_msgs.msg import Imu

def imuDataCallback(imu):
    rospy.loginfo("Quaternion Orientation:  [%f, %f, %f, %f]", imu.orientation.x, imu.orientation.y, imu.orientation.z, imu.orientation.w);
    rospy.loginfo("Angular Velocity:        [%f, %f, %f]", imu.angular_velocity.x, imu.angular_velocity.y, imu.angular_velocity.z);
    # add code here to handle incoming IMU data
    
def listener():

    rospy.init_node('listener', anonymous=True)

    # get the device name parameter
    deviceName = 'gx5'
    nameParam = rospy.get_name() + '/device'

    if rospy.has_param(nameParam):
        deviceName = rospy.get_param(nameParam)

        # clear parameter for future use
        rospy.delete_param(nameParam)

    # subscribe to the imu/data topic
    #Parameters:
    #   topic - namespace (defined in launch file) and topic name
    #   message type - type of message expected
    #   callback - callback function to handle this data
    rospy.Subscriber(("/" + deviceName + "/imu/data"), Imu, imuDataCallback)
    rospy.loginfo(("listening to /" + deviceName + "/imu/data"))

    # start listening for data
    rospy.spin()

if __name__ == '__main__':
    listener()

